### PR TITLE
Deprecating `SubTask.getLastBuiltOn`

### PR DESCRIPTION
**Pull request by jglick**
I was poking around in `SubTask`, noticed this method, and tried to find how it was used by the scheduler. So far as I can tell, it is not. It was introduced perhaps in d362c5609ddfc07feb43a740c6089f00bc64abaa or eedac7b0fcb4ca848111409817f8b431b5e77d6b (the history is hard to follow) but it seems that the usage was deleted the same day in 839ba1d772e889a0bd36b935100c49efb49f24aa.

### Testing done

None, just find usages within the repo.

### Proposed changelog entries

- N/A

Before the changes are marked as `ready-for-merge`:

```[tasklist]
### Maintainer checklist
- [x] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [x] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [x] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [x] Proper changelog labels are set so that the changelog can be generated automatically.
- [x] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [x] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).
```

### CODE REVIEW

**Review by NotMyFault — APPROVED** [review]

**Review by Vlatombe — APPROVED** [review]

**Review by MarkEWaite — APPROVED** [review]
This PR is now ready for merge.  We will merge it after approximately 24 hours if there is no negative feedback.

/label ready-for-merge
